src/main/java/graphql/execution/ValuesResolver.java

```diff
@@ -145,19 +146,23 @@ private Object coerceValue(VariableDefinition variableDefinition, GraphQLType gr
             return returnValue;
         }
 
-        if (value == null) return null;
+        if (value == null) {
+            return null;
+        }
 
         if (graphQLType instanceof GraphQLScalarType) {
             return coerceValueForScalar((GraphQLScalarType) graphQLType, value);
         } else if (graphQLType instanceof GraphQLEnumType) {
             return coerceValueForEnum((GraphQLEnumType) graphQLType, value);
         } else if (graphQLType instanceof GraphQLList) {
             return coerceValueForList(variableDefinition, (GraphQLList) graphQLType, value);
-        } else if (graphQLType instanceof GraphQLInputObjectType && value instanceof Map) {
-            //noinspection unchecked
-            return coerceValueForInputObjectType(variableDefinition, (GraphQLInputObjectType) graphQLType, (Map<String, Object>) value);
         } else if (graphQLType instanceof GraphQLInputObjectType) {
-            return value;
+            if (value instanceof Map) {
+                //noinspection unchecked
+                return coerceValueForInputObjectType(variableDefinition, (GraphQLInputObjectType) graphQLType, (Map<String, Object>) value);
+            } else {
+                throw new CoercingParseValueException("Variables for GraphQLInputObjectType must be an instance of a Map according to the graphql specification.  The offending object was a " + value.getClass().getName());
```

Comment:
Sorry for being picky but there's a double space after the full stop 😊

Reply:
Sorry I am old  and learnt to type  on mechanical typewriters and Mrs Cleary taught us that double spaces after a period are how you do it.  Its burnt into my muscle memory. ;)